taipy/gui/renderers/user_control.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2022 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from abc import ABC
+import typing as t
+import warnings
+
+from ..types import _AttributeType
+from .utils import _to_camel_case
+from .builder import _Builder
+
+if t.TYPE_CHECKING:
+    from ..gui import Gui
+
+
+class UserAttribute(ABC):
```

Comment:
Not an ABC class since it will be used directly by the user

Reply:
true that